core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java

```diff
@@ -96,6 +96,9 @@ public class MainClassBuildStep {
     static final String STARTUP_CONTEXT = "STARTUP_CONTEXT";
     static final String LOG = "LOG";
     static final String JAVA_LIBRARY_PATH = "java.library.path";
+    // This is declared as a constant so that it can be grepped for in the native-image binary using `strings`, e.g.:
+    // strings ./target/quarkus-runner | grep "quarkus.version="
+    public static final String GREPABLE_QUARKUS_VERSION = "GREPABLE_QUARKUS_VERSION";
```

Comment:
```suggestion
    public static final String GREPABLE_QUARKUS_VERSION = "__QUARKUS_VERSION_MARKER__";
```

not a fan of "grepable" :) so just tried to make one a big more explicit and unique.

Reply:
Just to make sure we are on the same page here before applying this.

What you need to `grep` for is `quarkus.version=` not `GREPABLE_QUARKUS_VERSION`.

`GREPABLE_QUARKUS_VERSION` is just the field name in the generated `io.quarkus.runner.ApplicationImpl` class.

@maxandersen Is that your understanding as well?

Reply:
I wonder if we should have something a bit more specific. To make sure we don't have two occurrences of the string at some point which could break the detection.

Maybe:

```
__quarkus_analytics__quarkus.version=
```

(to keep the analytics name we used previously)

And we could use this `__quarkus_analytics__` prefix for various things.

Reply:
And maybe we should have a specific class for all these future constants.

Now maybe it's too much, it's just ideas.

Reply:
No strong opinion on my side. It's definitely worth figuring out now since once shipped it won't be easy to change since people will adapt their tooling to depend on it.

Reply:
I think having a prefix is really something we want.
For the separate class, I would say it's an implementation detail.

I will let @maxandersen and @cescoffier chime in.

Reply:
I agree, having a prefix would prevent class and at least let us understand why we have this/these field/fields.

Reply:
> (to keep the analytics name we used previously)

@gsmet I didn't find any use of the `__quarkus_analytics__` prefix. Do you recall where it was used?

Reply:
@zakkak I didn't mean the prefix was used. I was talking about the `analytics` notion. That's how we called our data collection so I think it makes sense to be consistent with the naming.

Reply:
Maybe change the name of the `GREPABLE_QUARKUS_VERSION` constant too?

And I think you should go ahead with `__quarkus_analytics__quarkus.version` so that we can secure this feature for 3.15 LTS.

Reply:
Done

Reply:
just to be clear this variable is NOT related to build time analytics which we specifically have no interest of having runtime connection. this is simply for systems having way for other systems to separate a native image binary built with quarkus from other native images. 

Thats why I suggested `__QUARKUS_VERSION_MARKER__` as i thought thats whats "greppable".

isn't `__quarkus_version_marker_` sufficiently unique ?

Reply:
Chances are we will have more of these.

So having a common prefix is better. Thus why we went with `__quarkus_analytics__quarkus.version`.

If we add more, it's going to be easier to spot.